desktop-widgets/mainwindow.cpp

```diff
@@ -181,12 +183,21 @@ MainWindow::MainWindow() : QMainWindow(),
 							       { &plannerWidgets->plannerSettingsWidget, FLAG_NONE }, { &plannerWidgets->plannerDetails, FLAG_NONE } });
 	registerApplicationState(ApplicationState::EditPlannedDive, { { &plannerWidgets->plannerWidget, FLAG_NONE }, { profileContainer, FLAG_NONE },
 								      { diveList, FLAG_NONE },          { mapWidget, FLAG_NONE } });
-	registerApplicationState(ApplicationState::EditDiveSite, { { diveSiteEdit, FLAG_NONE }, { profileContainer, FLAG_DISABLED },
+	registerApplicationState(ApplicationState::EditDiveSite, { { diveSiteEdit.get(), FLAG_NONE }, { profileContainer, FLAG_DISABLED },
 								   { diveList, FLAG_DISABLED }, { mapWidget, FLAG_NONE } });
 	registerApplicationState(ApplicationState::FilterDive, { { mainTab.get(), FLAG_NONE }, { profileContainer, FLAG_NONE },
 								 { diveList, FLAG_NONE },      { &filterWidget, FLAG_NONE } });
 	registerApplicationState(ApplicationState::Statistics, { { statistics, FLAG_NONE }, { profileContainer, FLAG_NONE },
 								 { diveList, FLAG_DISABLED },   { &filterWidget, FLAG_NONE } });
+	registerApplicationState(ApplicationState::MapMaximized, { { nullptr, FLAG_NONE }, { nullptr, FLAG_NONE },
+								   { nullptr, FLAG_NONE }, { mapWidget, FLAG_NONE } });
+	registerApplicationState(ApplicationState::ProfileMaximized, { { nullptr, FLAG_NONE }, { profileContainer, FLAG_NONE },
+								       { nullptr, FLAG_NONE }, { nullptr, FLAG_NONE } });
+	registerApplicationState(ApplicationState::ListMaximized, { { nullptr, FLAG_NONE },  { nullptr, FLAG_NONE },
+								    { diveList, FLAG_NONE }, { nullptr, FLAG_NONE } });
+	registerApplicationState(ApplicationState::InfoMaximized, { { mainTab.get(), FLAG_NONE }, { nullptr, FLAG_NONE },
+								    { nullptr, FLAG_NONE },       { nullptr, FLAG_NONE } });
```

Comment:
since we are rethinking all of this - why does it make sense to implement the four full screen modes as states of the splitters. Maybe I'm missing the obvious problem, but couldn't we just simply hide the splitters and show those widgets as the only window content? Wouldn't that simplify things?

Reply:
> Maybe I'm missing the obvious problem, but couldn't we just simply hide the splitters and show those widgets as the only window content? Wouldn't that simplify things?

That's essentially what the code does. Only the splitters with content are shown. The advantage is that any combination is possible - from one to four widgets. For example in the statistics mode three widgets are shown: one on top, two at the bottom. You could also have one right, one left or one top, one bottom. I don't think adding an extra mode for a single widget would make things more simple..?